test/unit/esm-loader.test.mjs

```diff
@@ -60,6 +76,76 @@ tap.test('ES Module Loader', (t) => {
     td.reset()
   })
 
+  t.test('should not update the usage metric if misconfigured', async (t) => {
+    delete fakeNewrelic.agent
+
+    fakeGetOrCreateMetric.resetHistory()
+    fakeIncrementCallCount.resetHistory()
+    fakeSemver.gte.resetHistory()
+
+    // eslint-disable-next-line node/no-unsupported-features/es-syntax
+    loader = await import('../../esm-loader.mjs')
+
+    t.ok(fakeGetOrCreateMetric.notCalled, 'should not get a usage metric')
+    t.ok(fakeIncrementCallCount.notCalled, 'should not increment a usage metric')
+    t.ok(fakeSemver.gte.notCalled, 'should not have checked our node version')
+
+    t.end()
+  })
+
+  t.test('should update the loader metric when on a supported version of node', async (t) => {
+    fakeGetOrCreateMetric.resetHistory()
+    fakeIncrementCallCount.resetHistory()
+    fakeSemver.gte.resetHistory()
+
+    fakeSemver.gte.returns(true)
+    await td.replaceEsm('semver', {}, fakeSemver)
+
+    // eslint-disable-next-line node/no-unsupported-features/es-syntax
+    loader = await import('../../esm-loader.mjs')
+
+    t.ok(
+      fakeGetOrCreateMetric.calledOnceWith('Supportability/Features/ESM/Loader'),
+      'should get the correct usage metric'
+    )
+    t.equal(fakeIncrementCallCount.callCount, 1, 'should increment the metric')
+    t.ok(
+      fakeSemver.gte.calledOnceWith(process.version, 'v16.12.0'),
+      'should have checked our node version'
+    )
+
+    t.end()
+  })
+
+  t.test(
+    'should update the unsupported metric when on an unsupported version of node',
```

Comment:
this might be taking mocking to the max. it's fine but you can see in my unit tests i used the [skip attr](https://github.com/newrelic/node-newrelic/pull/1352/files#diff-51a8aeb8353bfdff3d3a85ecf4f3ada5b0463deedecdfba2f2b931dad5db2923R12) to only run only relevant versions.

Reply:
yep! i plan on merging your changes in and updating to follow that pattern instead of what i did here